rusk-wallet/src/bin/command.rs

```diff
@@ -932,20 +930,107 @@ pub enum RunResult<'a> {
     Tx(BlsScalar),
     PhoenixBalance(BalanceInfo, bool),
     MoonlightBalance(Dusk),
-    StakeInfo(StakeData, bool),
+    StakeInfo(Option<StakeData>, bool),
     Profile((u8, &'a Profile)),
     Profiles(&'a Vec<Profile>),
-    ContractId([u8; CONTRACT_ID_BYTES]),
+    ContractId(([u8; CONTRACT_ID_BYTES], u64)),
     ExportedKeys(PathBuf, PathBuf),
     Create(),
     Restore(),
     Settings(),
     PhoenixHistory(Vec<TransactionHistory>),
 }
 
-impl fmt::Display for RunResult<'_> {
+impl<'a> RunResult<'a> {
+    /// Convert the ran result into json
+    pub fn to_json(&self) -> String {
+        use RunResult::*;
+
+        match self {
+            PhoenixBalance(bal, _) => {
+                json!({
+                    "total": bal.value,
+                    "spendable": bal.spendable
+                })
+            }
+            MoonlightBalance(bal) => {
+                let bal = bal.to_string();
+
+                json!({
+                    "total": bal,
+                })
+            }
+            Profile((_, profile)) => profile.to_json(),
+            Profiles(addrs) => {
+                let addrs = addrs.iter().map(|a| a.to_json()).collect();
+
+                serde_json::Value::Array(addrs)
+            }
+            Tx(hash) => {
+                let hex = hex::encode(hash.to_bytes());
+
+                json!({
+                    "hash": hex
+                })
+            }
+            StakeInfo(Some(data), _) => match data.amount {
+                Some(amt) => {
+                    let amount = Dusk::from(amt.value).to_string();
+                    let locked = Dusk::from(amt.locked).to_string();
+                    let faults = data.faults.to_string();
+                    let hard_faults = data.hard_faults.to_string();
+                    let eligibility = amt.eligibility.to_string();
+                    let epoch = (amt.eligibility / EPOCH).to_string();
+                    let rewards = Dusk::from(data.reward).to_string();
+
+                    json!({
+                        "has_stake": true,
+                        "rewards": rewards,
+                        "amount": amount,
+                        "locked": locked,
+                        "eligibility": eligibility,
+                        "epoch": epoch,
+                        "hard_faults": hard_faults,
+                        "faults": faults,
+                    })
+                }
+                None => serde_json::Value::Null,
```

Comment:
this is not true, user can have rewards without a valid active stake 

to be more precise, data.amount can be none, but you can still have other info to show
```json

    {
        "amount": 0,
        "eligibility": 0,
        "faults": 0,
        "hard_faults": 0,
        "key": "mvft6AXjR1mDtg1Qo3GzGBKZAvvxY1S6h41DSEPh9Ewj9GFR61rPQnjVaM1W9nqkHijRC2d5Yo4FSBwU8wf9eHeEJYXuMSrB7eJxvsY6HjCeDJ9pKfJLdjayAfrxEyudvx2",
        "locked_amt": 0,
        "reward": 23261220305196
    },
```

Reply:
Addressed by https://github.com/dusk-network/rusk/pull/2774